### ISSUE TITLE
An item with the same key has already been added

### ISSUE THREAD

**Issue by molinch** [luberda-molinet/FFImageLoading#226]
@michaeldimoudis got this exception at the first start with 2.1.1. It never occurred afterwards.

```c#
2016-06-04 12:38:12.362 ContinuousFeedbackiOS[97190:6563997] Warning: Unhandled exception: System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. ---> System.ArgumentException: An item with the same key has already been added.
  at System.ThrowHelper.ThrowArgumentException (ExceptionResource resource) [0x00000] in /Users/builder/data/lanes/2922/7b89851d/source/maccore/_build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/src/mono/external/referencesource/mscorlib/system/throwhelper.cs:72 
  at System.Collections.Generic.Dictionary`2[TKey,TValue].Insert (System.Collections.Generic.TKey key, System.Collections.Generic.TValue value, Boolean add) [0x0008e] in /Users/builder/data/lanes/2922/7b89851d/source/maccore/_build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/src/mono/external/referencesource/mscorlib/system/collections/generic/dictionary.cs:336 
  at System.Collections.Generic.Dictionary`2[TKey,TValue].Add (System.Collections.Generic.TKey key, System.Collections.Generic.TValue value) [0x00000] in /Users/builder/data/lanes/2922/7b89851d/source/maccore/_build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/src/mono/external/referencesource/mscorlib/system/collections/generic/dictionary.cs:192 
  at FFImageLoading.Work.WorkScheduler+<QueueTaskAsync>d__25.MoveNext () [0x000ab] in <filename unknown>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in /Users/builder/data/lanes/2922/7b89851d/source/maccore/_build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/src/mono/external/referencesource/mscorlib/system/runtime/exceptionservices/exceptionservicescommon.cs:143 
  at FFImageLoading.Work.WorkScheduler+<QueueTaskAsync>d__25.MoveNext () [0x002bc] in <filename unknown>:0 
  --- End of inner exception stack trace ---
---> (Inner Exception #0) System.ArgumentException: An item with the same key has already been added.
  at System.ThrowHelper.ThrowArgumentException (ExceptionResource resource) [0x00000] in /Users/builder/data/lanes/2922/7b89851d/source/maccore/_build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/src/mono/external/referencesource/mscorlib/system/throwhelper.cs:72 
  at System.Collections.Generic.Dictionary`2[TKey,TValue].Insert (System.Collections.Generic.TKey key, System.Collections.Generic.TValue value, Boolean add) [0x0008e] in /Users/builder/data/lanes/2922/7b89851d/source/maccore/_build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/src/mono/external/referencesource/mscorlib/system/collections/generic/dictionary.cs:336 
  at System.Collections.Generic.Dictionary`2[TKey,TValue].Add (System.Collections.Generic.TKey key, System.Collections.Generic.TValue value) [0x00000] in /Users/builder/data/lanes/2922/7b89851d/source/maccore/_build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/src/mono/external/referencesource/mscorlib/system/collections/generic/dictionary.cs:192 
  at FFImageLoading.Work.WorkScheduler+<QueueTaskAsync>d__25.MoveNext () [0x000ab] in <filename unknown>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in /Users/builder/data/lanes/2922/7b89851d/source/maccore/_build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/src/mono/external/referencesource/mscorlib/system/runtime/exceptionservices/exceptionservicescommon.cs:143 
  at FFImageLoading.Work.WorkScheduler+<QueueTaskAsync>d__25.MoveNext () [0x002bc] in <filename unknown>:0 <---
```

**Comment by NVentimiglia** [comment]
I see the fix, when should we expect a merge and nuget package ?

**Comment by daniel-luberda** [comment]
Should be soon. If you need to fix it ASAP you can use nugets from our AppVeyor page,  where nugets are build for every commit.

**Comment by NVentimiglia** [comment]
@daniel-luberda  Great, can I get a link ?

**Comment by daniel-luberda** [comment]
It's on project page (just click Build badge)  https://ci.appveyor.com/project/molinch/ffimageloading/build/artifacts

**Comment by daniel-luberda** [comment]
Nuget with fix released (2.1.2)
